.github/workflows/pr-go.yaml

```diff
@@ -0,0 +1,138 @@
+
+env:
+  PROTOC_VERSION: 3.18.1
+  BUILDIFIER_VERSION: 5.1.0
+  GO_VERSION: 1.17.0
+  GOLANGCI_LINT: v1.46.2
+  PROTOLOCK_VERSION: v0.15.0
+  PROTOC_GEN_GO_VERSION: v1.5.2
+
+jobs:
+  unit-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: bazelbuild/setup-bazelisk@v2.0.0
+      - name: Mount bazel cache
+        uses: actions/cache@v3
+        with:
+          path: "~/.cache/bazel"
+          key: bazel
+      - name: Run unit test
+        run: make test
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: bazelbuild/setup-bazelisk@v2.0.0
+      - name: Mount bazel cache
+        uses: actions/cache@v3
+        with:
+          path: "~/.cache/bazel"
+          key: bazel
+      - name: Build
+        run: make build
+
+  gofmt-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Install goimports
+        run: go get golang.org/x/tools/cmd/goimports
+      - name: Check Go formatting
+        run: make gofmt-check
+
+  proto-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install protolock
+        run: go install github.com/nilslice/protolock/cmd/protolock@${{ env.PROTOLOCK_VERSION }}
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          version: ${{ env.PROTOC_VERSION }}
+      - name: Check proto files
+        run: make proto-check
+
+  buildifier-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check Bazel file formatting
+        uses: thompsonja/bazel-buildifier@v0.4.0
+        with:
+          warnings: -function-docstring,-function-docstring-header,-function-docstring-args,-function-docstring-return,-module-docstring,-skylark-docstring,-rule-impl-return
+          buildifier_version: ${{ env.BUILDIFIER_VERSION }}
+
+  gazelle-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: bazelbuild/setup-bazelisk@v2.0.0
+      - name: Check build files
+        run: make gazelle-check
+
+  update-repos-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Set GOPATH env
+        run: echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+      - name: Clone googleapis repository
+        run: git clone https://github.com/googleapis/googleapis.git $GOPATH/src/github.com/googleapis/googleapis
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          version: ${{ env.PROTOC_VERSION }}
+      - name: Install protoc-gen-go
+        run: go install github.com/golang/protobuf/protoc-gen-go@${{ env.PROTOC_GEN_GO_VERSION }}
+      - name: Generate proto go files
```

Comment:
Because the `update-repos-check` calls `go mod tidy`, we need to generate the proto-go files.